src/dlsnode/neo/request/GetRange.d

```diff
@@ -341,27 +341,30 @@ private scope class GetRangeImpl_v0: GetRangeProtocol_v0
     override protected bool getNextRecord ( out time_t timestamp, ref void[] value,
             out bool wait_for_data)
     {
+record_loop:
         for (;;)
         {
-            bool last_key = this.iterator.next(this.job_notification, timestamp, value,
-                    wait_for_data);
+            auto res = this.iterator.next(this.job_notification, timestamp, value);
 
-            if (wait_for_data)
+            with (NeoStorageEngineStepIterator.NextResult)
             {
-                return true;
-            }
-
-            if (last_key)
-            {
-                return false;
-            }
-
-            if (!this.filterRecord(timestamp, value))
-            {
-                continue;
+                switch (res)
+                {
+                    case NoMoreData:
+                        return false;
+                    case WaitForData:
+                        wait_for_data = true;
+                        return true;
+                    case RecordRead:
+                        if (!this.filterRecord(timestamp, value))
+                        {
+                            continue record_loop;
```

Comment:
Out of interest: is the label needed?

Reply:
Puh, it could be just `break`.

Reply:
Oh, haha, I forget about my comment and just merged. Feel free to make a small PR changing this, if it makes sense. If it were `break`, then the label _would_ be needed btw.

Reply:
No it wouldn't be needed, because we would just go to the next iteration step, jumping out of `switch` :-)

https://github.com/sociomantic-tsunami/dlsnode/blob/neo/src/dlsnode/neo/request/GetRange.d#L344-L367

(notice no `return` after `switch`).